WooCommerce/Classes/ViewRelated/Editor/AztecEditorViewController.swift

```diff
@@ -0,0 +1,863 @@
+            optionsTablePresenter.dismiss()
+            return
+        }
+
+//        trackFormatBarAnalytics(stat: .editorTappedHeader)
+
+        let headerOptions = Constants.headers.map { headerType -> OptionsTableViewOption in
+            let attributes: [NSAttributedString.Key: Any] = [
+                .font: UIFont.systemFont(ofSize: CGFloat(headerType.fontSize)),
+            ]
+
+            let title = NSAttributedString(string: headerType.description, attributes: attributes)
+
+            return OptionsTableViewOption(image: headerType.iconImage,
+                                          title: title,
+                                          accessibilityLabel: headerType.accessibilityLabel)
+        }
+
+        let selectedIndex = Constants.headers.firstIndex(of: self.headerLevelForSelectedText())
+
+        let optionsTableViewController = OptionsTableViewController(options: headerOptions)
+        optionsTableViewController.applyDefaultStyles()
+
+        optionsTablePresenter.present(
+            optionsTableViewController,
+            fromBarItem: item,
+            selectedRowIndex: selectedIndex,
+            onSelect: { [weak self] selected in
+                guard let range = self?.richTextView.selectedRange else { return }
+
+//                let selectedStyle = Analytics.headerStyleValues[selected]
+//                self?.trackFormatBarAnalytics(stat: .editorTappedHeaderSelection, headingStyle: selectedStyle)
+
+                self?.richTextView.toggleHeader(Constants.headers[selected], range: range)
+                self?.optionsTablePresenter.dismiss()
+        })
+    }
+
+    func insertHorizontalRuler() {
+//        trackFormatBarAnalytics(stat: .editorTappedHorizontalRule)
+        richTextView.replaceWithHorizontalRuler(at: richTextView.selectedRange)
+    }
+
+    func insertMore() {
+//        trackFormatBarAnalytics(stat: .editorTappedMore)
+        richTextView.replace(richTextView.selectedRange, withComment: Constants.moreAttachmentText)
+    }
+
+    func headerLevelForSelectedText() -> Header.HeaderType {
+        var identifiers = Set<FormattingIdentifier>()
+        if richTextView.selectedRange.length > 0 {
+            identifiers = richTextView.formattingIdentifiersSpanningRange(richTextView.selectedRange)
+        } else {
+            identifiers = richTextView.formattingIdentifiersForTypingAttributes()
+        }
+        let mapping: [FormattingIdentifier: Header.HeaderType] = [
+            .header1: .h1,
+            .header2: .h2,
+            .header3: .h3,
+            .header4: .h4,
+            .header5: .h5,
+            .header6: .h6,
+        ]
+        for (key, value) in mapping {
+            if identifiers.contains(key) {
+                return value
+            }
+        }
+        return .none
+    }
+}
+
+extension AztecEditorViewController {
+    // MARK: - Keyboard Handling
+
+    override var keyCommands: [UIKeyCommand] {
```

Comment:
This could also be extracted to a separate entity, let's call it KeyboardManager? It could be the thing that encapsulates everything related to keyboard management